05/app.js

```diff
@@ -0,0 +1,82 @@
+const formEl = document.querySelector('form');
+
+const labelList = document.querySelectorAll('label');
+
+formEl.addEventListener('submit', validateData);
+
+function validateData(e) {
+	e.preventDefault();
+
+	const firstName = e.target.elements.firstName.value;
+	const lastName = e.target.elements.lastName.value;
+	const avoidChars = /[1-9!@#$%^&*]/g;
+	fullName(firstName, lastName, avoidChars);
+
+	const streetName = e.target.elements.street.value;
+	streetNameValidation(streetName);
+
+	const houseNumber = e.target.elements.houseNumber.value;
+	const flatNumber = e.target.elements.flatNumber.value;
+	buildingNumber(houseNumber, flatNumber);
+
+	const city = e.target.elements.city.value;
+	cityValidation(city, avoidChars);
+
+	finalValidation();
```

Comment:
Co z kodem pocztowym? Zawsze jest poprawny?

Reply:
Z kodem pocztowym praktycznie taka sama sytuacja, jednakże posiada on dodatkowo atrybut pattern, który wymusza na użytkowniku wpisanie poprawnego formatu.

Reply:
Jasne, ale walidację przez użytkownika łatwo można wyłączyć i wtedy nie mamy już zabezpieczenia przed nieprawidłowymi danymi :) Pamiętaj też, że nie wszystkie przeglądarki muszą to wspierać np. Opera Mini (https://caniuse.com/input-pattern).
Po prostu zawsze powinniśmy mieć plan B na wypadek, gdy przeglądarka nie zadziała jak trzeba ;)